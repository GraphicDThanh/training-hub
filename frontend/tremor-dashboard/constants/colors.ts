type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const COLOR_CHANNELS_CHART: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};
