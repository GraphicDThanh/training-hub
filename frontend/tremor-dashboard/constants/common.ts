// Types
import { OptionType } from "@/types";

export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export enum DIRECTION {
  ASC = "asc",
  DESC = "desc",
}

// Mailto Support
export const MAILTO_SUPPORT = "support@tremor-dashboard.com";

export const REMEMBER_ME_COOKIES_KEY = "remember-me";
export const UID_KEY = "uid";
export const ROLE = "role";
export const NAME = "name";
export const AVATAR = "avatar";
export const AUTH_SESSION_COOKIES_KEY = "authjs.session-token";

export const ORDER_LIST_OPTIONS: OptionType[] = [
  { option: "Paid", value: "0" },
  { option: "Refunded", value: "2" },
  { option: "Canceled", value: "1" },
];

export const PRODUCT_LIST_OPTIONS: OptionType[] = [
  { option: "Yes", value: "true" },
  { option: "No", value: "false" },
];

export const FORMAT_DATE = {
  YEAR_MONTH_DATE_TIME: "YYYY-MM-DDTHH:mm:ss.SSS",
  MONTH_DAY_YEAR_TIME: "MMM D, YYYY h:mm A",
};

export const SEPARATOR = {
  COMMAS: ",",
  DOTS: ".",
};

export const CDN_KEY = process.env.NEXT_PUBLIC_CDN_KEY;

export const DRAG_ZONE = {
  DEFAULT: {
    TEXT: "Drag and drop your file here to begin uploading.",
    STYLE_INPUT: "border-gray-300",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_DRAG: {
    TEXT: "Release the mouse click to upload the photo",
    STYLE_INPUT: "outline outline-2 outline-[#ebeff4]",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_WRONG_FORMAT: {
    TEXT: "Invalid image file type. Please try uploading again.",
    STYLE_INPUT: "border-red-500",
    STYLE_TEXT: "text-red-500 dark:text-red-500",
  },
};

export const TRANSACTION_CLASS = {
  INCREASE: {
    buttonClasses: "text-few dark:text-few dark:hove:text-few border-few",
    textClasses: "font-semibold text-greener dark:text-few",
  },
  DECREASE: {
    buttonClasses:
      "text-attention dark:text-attention dark:hover:text-attention border-attention",
    textClasses: "font-semibold text-blood dark:text-pink",
  },
  PENDING: {
    buttonClasses:
      "text-primary dark:text-primary dark:hover:text-primary border-primary",
    textClasses:
      "font-semibold text-primary dark:text-dark-tremor-content-pending",
  },
};

export const enum VARIANT_BUTTON {
  LIGHT = "light",
  LIGHT_CARD = "lightCard",
  LIGHT_CENTER = "lightCenter",
  LIGHT_STATUS = "lightStatus",
  SECONDARY = "secondary",
  SECONDARY_SHADOW = "secondaryShadow",
  PRIMARY = "primary",
  PRIMARY_CENTER = "primaryCenter",
  SURFACE = "surface",
  DARK = "dark",
  ACTIVE = "active",
}

export const NO_IMAGE = "No Image";

export const RESULT_NOT_FOUND = "Result Not Found";

export const AMOUNT_LIMIT = 20000;
export const PRICE_LIMIT = 20000;
export const QUANTITY_LIMIT = 10000;
export const WEIGHT_LIMIT = 2000;

export const PREFIX_PHONE_NUMBER = "+1";

export const STORAGES = {
  IS_DARK_THEME: "isDarkTheme",
};

export const THEMES = {
  LIGHT: "light",
  DARK: "dark",
};

export const LOGO_TITLE = "Tremor Dashboard PRO";
