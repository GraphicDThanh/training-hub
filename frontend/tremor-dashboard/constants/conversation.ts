export const CONVERSATION_TABLES = {
  GROUPS: "groups",
  CHATS: "chats",
};

export const CONVERSATION_ORDERS = {
  CREATED_AT: "createdAt",
  UPDATED_AT: "updatedAt",
};

export const CONVERSATION_QUERYS = {
  USER_IDS: "userIds",
};

export enum CONVERSATION_STATUS {
  INITIAL = "initial",
  FETCHED = "fetched",
}

export const ADMIN_ID = parseInt(process.env.NEXT_PUBLIC_ADMIN_ID!);

export enum DOCUMENT_CHANGE_TYPE {
  ADDED = "added",
  MODIFIED = "modified",
  REMOVED = "removed",
}

export enum WHERE_FILTER_OP {
  ARRAY_CONTAINS = "array-contains",
}
