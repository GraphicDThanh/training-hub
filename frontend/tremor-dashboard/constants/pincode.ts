export const PIN_CODE_DEFAULT = "*";
export const PIN_CODE_LENGTH = 6;
export const PIN_CODE_SKIP_SET_KEY = "skipSetupPinCode";
