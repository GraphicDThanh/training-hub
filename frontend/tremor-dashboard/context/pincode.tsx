"use client";

import {
  ReactNode,
  createContext,
  useCallback,
  useContext,
  useMemo,
  useState,
} from "react";

import { updatePinCode } from "@/services";

interface IPinCodeContext {
  pinCode?: string;
  setPinCode: (value: string) => void;
}

const initialPinCodeContext: IPinCodeContext = {
  setPinCode: () => {},
};

const PinCodeContext = createContext(initialPinCodeContext);

interface IPinCodeProvider {
  children: ReactNode;
  pinCode?: string;
}

const PinCodeProvider = ({
  children,
  pinCode: initialPinCode,
}: IPinCodeProvider) => {
  const [pinCode, setPinCode] = useState<string | undefined>(initialPinCode);

  const pinCodeContextValue: IPinCodeContext = useMemo(
    () => ({ pinCode, setPinCode }),
    [pinCode],
  );

  return (
    <PinCodeContext.Provider value={pinCodeContextValue}>
      {children}
    </PinCodeContext.Provider>
  );
};

const usePinCode = () => {
  const context = useContext(PinCodeContext);

  if (!context) {
    throw new Error("usePinCode hooks should using inside PinCodeProvider!");
  }

  const { pinCode, setPinCode } = context;

  const handleSetPinCode = useCallback(
    async (pinCode: string) => {
      const { errorMessage } = await updatePinCode(pinCode);

      if (errorMessage) {
        setPinCode(pinCode);
      }

      return !!errorMessage;
    },
    [setPinCode],
  );

  return {
    pinCode,
    setPinCode: handleSetPinCode,
  };
};

export { PinCodeProvider, usePinCode };
