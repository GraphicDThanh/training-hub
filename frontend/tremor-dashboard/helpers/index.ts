export * from "./formatDate";
export * from "./formatFormData";
export * from "./common";
export * from "./format";
export * from "./formatNumber";
export * from "./billing";
export * from "./apis";
export * from "./pincode";
export * from "./time";
export * from "./user";
export * from "./project";
export * from "./validate";
export * from "./colorScheme";
export * from "./styles";
