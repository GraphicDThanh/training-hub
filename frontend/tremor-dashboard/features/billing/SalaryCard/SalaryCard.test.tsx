import { fireEvent, render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";

// Mocks
import { MOCK_SALARY_DATA } from "@/mocks";

// Hocs
import { TWithPinCode } from "@/hocs/withPinCode";

// Types
import { SalaryCardData } from "@/types";

// Contexts
import { PinCodeProvider } from "@/context/pincode";

// Constants
import { KEYBOARD_KEYS } from "@/constants";

// Helpers
import { moneyFormat } from "@/helpers";

// Components
import { SalaryCard } from "..";

class ResizeObserver {
  observe() {}
  unobserve() {}
  disconnect() {}
}
window.ResizeObserver = ResizeObserver;

const setup = (overrideProps?: Partial<TWithPinCode<SalaryCardData>>) => {
  return render(
    <PinCodeProvider pinCode="111111">
      <SalaryCard {...MOCK_SALARY_DATA[0]} {...overrideProps} />
    </PinCodeProvider>,
  );
};

const { DIGIT_1 } = KEYBOARD_KEYS;

describe("SalaryCard Testing", () => {
  it("Should match snapshot", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("Should open PIN code modal when isShowAmount is false and toggleShowHide", async () => {
    const { getByTestId } = setup({
      type: MOCK_SALARY_DATA[1].type,
    });

    fireEvent.click(getByTestId("btn-eyes"));

    const modal = await waitFor(() => getByTestId("modal-pincode"));
    expect(modal).toBeTruthy();
    expect(modal).toMatchSnapshot();
  });

  it("Should submit success when typing pin code correctly", async () => {
    const { getByTestId, getAllByTestId, getByText, container } = setup({
      value: 14444,
    });

    fireEvent.click(getByTestId("btn-eyes"));

    await waitFor(() => getByTestId("modal-pincode"));

    const clickPinCodeField = (index: number) => {
      waitFor(() =>
        fireEvent.keyDown(getAllByTestId("pin-code-input")[index], {
          key: DIGIT_1,
        }),
      );
    };

    clickPinCodeField(0);
    clickPinCodeField(1);
    clickPinCodeField(2);
    clickPinCodeField(3);
    clickPinCodeField(4);
    clickPinCodeField(5);

    expect(getByText("Submit")).not.toBeDisabled();
    fireEvent.click(getByText("Submit"));

    expect(container).toMatchSnapshot();

    expect(
      getByText(
        moneyFormat({
          value: 14444,
          currency: "$",
        }),
      ),
    ).toBeTruthy();

    fireEvent.click(getByTestId("btn-eyes"));

    expect(getByText("*****")).toBeInTheDocument;
  });

  it("Should open PIN code modal when isShowAmount is false and toggleShowHide", async () => {
    const { getByTestId } = setup({
      type: MOCK_SALARY_DATA[1].type,
    });

    fireEvent.click(getByTestId("btn-eyes"));

    const modal = await waitFor(() => getByTestId("modal-pincode"));
    expect(modal).toBeTruthy();
    expect(modal).toMatchSnapshot();
  });
});
