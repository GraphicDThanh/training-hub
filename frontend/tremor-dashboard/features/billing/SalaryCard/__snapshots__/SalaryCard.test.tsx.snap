// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SalaryCard Testing Should match snapshot 1`] = `
<div>
  <div
    class="w-full font-primary antialiased items-center justify-between"
  >
    <div
      class="flex items-center"
    >
      <div
        class="tremor-Card-root w-full text-left rounded-tremor-default p-6 ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand bg-tremor-primary dark:bg-dark-tremor-primary mx-auto pt-3 pb-4 px-4 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative rounded-xl shadow-md md:max-h-[236px] md:min-h-[236px]"
      >
        <div
          class="tremor-Flex-root flex w-full flex-col justify-between items-center"
        >
          <div
            class="tremor-Flex-root flex flex-row items-center p-1 justify-center rounded-xl shadow-box-icon-default w-16 h-16 bg-gradient-pickled"
          >
            <svg
              color="white"
              fill="currentColor"
              height="23px"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 24 24"
              width="23px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M4 10h3v7H4zM10.5 10h3v7h-3zM2 19h20v3H2zM17 10h3v7h-3zM12 1L2 6v2h20V6z"
              />
            </svg>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-center mt-4"
          >
            <p
              class="mb-1 text-primary dark:text-lighter text-tremor-title leading-[33px] tracking-[0.1764px] font-semibold"
            >
              Salary
            </p>
            <div
              class="tremor-Flex-root flex w-full flex-row justify-center items-center min-h-[35px]"
            >
              <p
                class="text-tremor-default text-xs text-center dark:text-dark-romance text-primary font-light"
              >
                Belong Interactive
              </p>
            </div>
          </div>
        </div>
        <div
          class="tremor-Divider-root w-full mx-auto flex justify-between gap-3 items-center text-tremor-default text-tremor-content dark:text-dark-tremor-content opacity-25 dark:opacity-15 my-3"
        >
          <div
            class="w-full h-[1px] bg-tremor-border dark:bg-dark-tremor-border"
          />
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center gap-2 min-h-[35px]"
        >
          <p
            class="text-tremor-default text-primary text-xl h-[20px] dark:text-lighter font-semibold min"
          >
            *****
          </p>
          <button
            aria-label="Toggle Show Salary Button"
            class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none text-sm bg-transparent text-primary hover:text-primary dark:text-lighter dark:hover:text-lighter"
            data-testid="btn-eyes"
          >
            <svg
              class="tremor-Button-icon shrink-0 h-5 w-5"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M288 80c-65.2 0-118.8 29.6-159.9 67.7C89.6 183.5 63 226 49.4 256c13.6 30 40.2 72.5 78.6 108.3C169.2 402.4 222.8 432 288 432s118.8-29.6 159.9-67.7C486.4 328.5 513 286 526.6 256c-13.6-30-40.2-72.5-78.6-108.3C406.8 109.6 353.2 80 288 80zM95.4 112.6C142.5 68.8 207.2 32 288 32s145.5 36.8 192.6 80.6c46.8 43.5 78.1 95.4 93 131.1c3.3 7.9 3.3 16.7 0 24.6c-14.9 35.7-46.2 87.7-93 131.1C433.5 443.2 368.8 480 288 480s-145.5-36.8-192.6-80.6C48.6 356 17.3 304 2.5 268.3c-3.3-7.9-3.3-16.7 0-24.6C17.3 208 48.6 156 95.4 112.6zM288 336c44.2 0 80-35.8 80-80s-35.8-80-80-80c-.7 0-1.3 0-2 0c1.3 5.1 2 10.5 2 16c0 35.3-28.7 64-64 64c-5.5 0-10.9-.7-16-2c0 .7 0 1.3 0 2c0 44.2 35.8 80 80 80zm0-208a128 128 0 1 1 0 256 128 128 0 1 1 0-256z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SalaryCard Testing Should open PIN code modal when isShowAmount is false and toggleShowHide 1`] = `
<div
  aria-modal="true"
  class="tremor-dialog-root relative z-50"
  data-headlessui-state="open"
  data-testid="modal-pincode"
  id="pincode-modal"
  role="dialog"
>
  <div
    class="fixed inset-0 bg-slate-950/30 dark:bg-slate-950/50 transition-opacity ease-out duration-300 opacity-0"
  />
  <div
    class="fixed inset-0 overflow-y-auto w-screen"
  >
    <div
      class="flex min-h-full items-center justify-center p-4"
    >
      <div
        class="tremor-dialog-panel max-w-lg overflow-hidden text-left ring-1 shadow-tremor transition-all transform text-tremor-content ring-tremor-ring dark:text-dark-tremor-content dark:ring-dark-tremor-ring rounded-tremor-default p-6 rounded-md border-none outline-none flex flex-col gap-8 sm:min-w-[80%] bg-tremor-primary dark:bg-dark-tremor-primary md:min-w-[390px] w-[calc(100%-8px)] md:max-w-[390px] ease-out duration-300 opacity-0 scale-95"
        data-headlessui-state="open"
        id="headlessui-dialog-panel-:r5:"
      >
        <h2
          class="text-primary dark:text-white font-bold text-2xl text-center"
        >
          Enter your PIN code
        </h2>
        <div>
          <div
            class="tremor-Flex-root flex flex-row justify-between items-center gap-3 sm:gap-6 w-full self-center max-w-[250px] sm:max-w-[360px] m-auto"
          >
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent focused"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-evenly items-center gap-5"
        >
          <button
            class="tremor-Button-root inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted border-tremor-brand dark:border-dark-tremor-brand hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis opacity-50 cursor-not-allowed py-3 px-6 rounded-lg border-none bg-gradient-primary dark:bg-gradient-pickled shadow-btn-primary dark:!shadow-btn-primary hover:!shadow-btn-primary-hover dark:hover:!shadow-btn-primary-hover flex-1 min-w-[64px] sm:px-[22px]"
            disabled=""
          >
            <span
              class="tremor-Button-text text-tremor-default whitespace-nowrap"
            >
              <p
                class="text-tremor-default flex items-center uppercase py-[2px] text-xs font-bold text-white dark:text-dark-tremor-content-title tracking-wide"
              >
                Submit
              </p>
            </span>
          </button>
          <button
            class="tremor-Button-root inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis py-3 px-6 rounded-lg border-none bg-gradient-btn-back dark:bg-gradient-btn-back shadow-btn-surface dark:!shadow-btn-surface hover:!shadow-btn-surface-hover dark:hover:!shadow-btn-surface-hover mt-0 flex-1"
            data-testid="btn-close-modal"
            type="submit"
          >
            <span
              class="tremor-Button-text text-tremor-default whitespace-nowrap"
            >
              <p
                class="text-tremor-default uppercase font-bold text-xs text-gray-900 dark:text-black tracking-wide"
              >
                Cancel
              </p>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SalaryCard Testing Should open PIN code modal when isShowAmount is false and toggleShowHide 2`] = `
<div
  aria-modal="true"
  class="tremor-dialog-root relative z-50"
  data-headlessui-state="open"
  data-testid="modal-pincode"
  id="pincode-modal"
  role="dialog"
>
  <div
    class="fixed inset-0 bg-slate-950/30 dark:bg-slate-950/50 transition-opacity ease-out duration-300 opacity-0"
  />
  <div
    class="fixed inset-0 overflow-y-auto w-screen"
  >
    <div
      class="flex min-h-full items-center justify-center p-4"
    >
      <div
        class="tremor-dialog-panel max-w-lg overflow-hidden text-left ring-1 shadow-tremor transition-all transform text-tremor-content ring-tremor-ring dark:text-dark-tremor-content dark:ring-dark-tremor-ring rounded-tremor-default p-6 rounded-md border-none outline-none flex flex-col gap-8 sm:min-w-[80%] bg-tremor-primary dark:bg-dark-tremor-primary md:min-w-[390px] w-[calc(100%-8px)] md:max-w-[390px] ease-out duration-300 opacity-0 scale-95"
        data-headlessui-state="open"
        id="headlessui-dialog-panel-:rl:"
      >
        <h2
          class="text-primary dark:text-white font-bold text-2xl text-center"
        >
          Enter your PIN code
        </h2>
        <div>
          <div
            class="tremor-Flex-root flex flex-row justify-between items-center gap-3 sm:gap-6 w-full self-center max-w-[250px] sm:max-w-[360px] m-auto"
          >
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent focused"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-evenly items-center gap-5"
        >
          <button
            class="tremor-Button-root inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted border-tremor-brand dark:border-dark-tremor-brand hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis opacity-50 cursor-not-allowed py-3 px-6 rounded-lg border-none bg-gradient-primary dark:bg-gradient-pickled shadow-btn-primary dark:!shadow-btn-primary hover:!shadow-btn-primary-hover dark:hover:!shadow-btn-primary-hover flex-1 min-w-[64px] sm:px-[22px]"
            disabled=""
          >
            <span
              class="tremor-Button-text text-tremor-default whitespace-nowrap"
            >
              <p
                class="text-tremor-default flex items-center uppercase py-[2px] text-xs font-bold text-white dark:text-dark-tremor-content-title tracking-wide"
              >
                Submit
              </p>
            </span>
          </button>
          <button
            class="tremor-Button-root inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis py-3 px-6 rounded-lg border-none bg-gradient-btn-back dark:bg-gradient-btn-back shadow-btn-surface dark:!shadow-btn-surface hover:!shadow-btn-surface-hover dark:hover:!shadow-btn-surface-hover mt-0 flex-1"
            data-testid="btn-close-modal"
            type="submit"
          >
            <span
              class="tremor-Button-text text-tremor-default whitespace-nowrap"
            >
              <p
                class="text-tremor-default uppercase font-bold text-xs text-gray-900 dark:text-black tracking-wide"
              >
                Cancel
              </p>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SalaryCard Testing Should submit success when typing pin code correctly 1`] = `
<div>
  <div
    class="w-full font-primary antialiased items-center justify-between"
  >
    <div
      class="flex items-center"
    >
      <div
        class="tremor-Card-root w-full text-left rounded-tremor-default p-6 ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand bg-tremor-primary dark:bg-dark-tremor-primary mx-auto pt-3 pb-4 px-4 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative rounded-xl shadow-md md:max-h-[236px] md:min-h-[236px]"
      >
        <div
          class="tremor-Flex-root flex w-full flex-col justify-between items-center"
        >
          <div
            class="tremor-Flex-root flex flex-row items-center p-1 justify-center rounded-xl shadow-box-icon-default w-16 h-16 bg-gradient-pickled"
          >
            <svg
              color="white"
              fill="currentColor"
              height="23px"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 24 24"
              width="23px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M4 10h3v7H4zM10.5 10h3v7h-3zM2 19h20v3H2zM17 10h3v7h-3zM12 1L2 6v2h20V6z"
              />
            </svg>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-center mt-4"
          >
            <p
              class="mb-1 text-primary dark:text-lighter text-tremor-title leading-[33px] tracking-[0.1764px] font-semibold"
            >
              Salary
            </p>
            <div
              class="tremor-Flex-root flex w-full flex-row justify-center items-center min-h-[35px]"
            >
              <p
                class="text-tremor-default text-xs text-center dark:text-dark-romance text-primary font-light"
              >
                Belong Interactive
              </p>
            </div>
          </div>
        </div>
        <div
          class="tremor-Divider-root w-full mx-auto flex justify-between gap-3 items-center text-tremor-default text-tremor-content dark:text-dark-tremor-content opacity-25 dark:opacity-15 my-3"
        >
          <div
            class="w-full h-[1px] bg-tremor-border dark:bg-dark-tremor-border"
          />
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center gap-2 min-h-[35px]"
        >
          <div
            class="tremor-Flex-root flex w-full flex-row justify-start items-center"
          >
            <p
              class="text-tremor-default text-primary text-xl dark:text-lighter font-semibold"
            >
              +
            </p>
            <p
              class="text-tremor-default text-primary !text-lg dark:text-lighter font-semibold truncate max-w-[89px]"
            >
              $14,444.00
            </p>
          </div>
          <button
            aria-label="Toggle Show Salary Button"
            class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none text-sm bg-transparent text-primary hover:text-primary dark:text-lighter dark:hover:text-lighter"
            data-testid="btn-eyes"
          >
            <svg
              class="tremor-Button-icon shrink-0 h-5 w-5"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 640 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M38.8 5.1C28.4-3.1 13.3-1.2 5.1 9.2S-1.2 34.7 9.2 42.9l592 464c10.4 8.2 25.5 6.3 33.7-4.1s6.3-25.5-4.1-33.7L525.6 386.7c39.6-40.6 66.4-86.1 79.9-118.4c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C465.5 68.8 400.8 32 320 32c-68.2 0-125 26.3-169.3 60.8L38.8 5.1zm151 118.3C226 97.7 269.5 80 320 80c65.2 0 118.8 29.6 159.9 67.7C518.4 183.5 545 226 558.6 256c-12.6 28-36.6 66.8-70.9 100.9l-53.8-42.2c9.1-17.6 14.2-37.5 14.2-58.7c0-70.7-57.3-128-128-128c-32.2 0-61.7 11.9-84.2 31.5l-46.1-36.1zM394.9 284.2l-81.5-63.9c4.2-8.5 6.6-18.2 6.6-28.3c0-5.5-.7-10.9-2-16c.7 0 1.3 0 2 0c44.2 0 80 35.8 80 80c0 9.9-1.8 19.4-5.1 28.2zm9.4 130.3C378.8 425.4 350.7 432 320 432c-65.2 0-118.8-29.6-159.9-67.7C121.6 328.5 95 286 81.4 256c8.3-18.4 21.5-41.5 39.4-64.8L83.1 161.5C60.3 191.2 44 220.8 34.5 243.7c-3.3 7.9-3.3 16.7 0 24.6c14.9 35.7 46.2 87.7 93 131.1C174.5 443.2 239.2 480 320 480c47.8 0 89.9-12.9 126.2-32.5l-41.9-33zM192 256c0 70.7 57.3 128 128 128c13.3 0 26.1-2 38.2-5.8L302 334c-23.5-5.4-43.1-21.2-53.7-42.3l-56.1-44.2c-.2 2.8-.3 5.6-.3 8.5z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
